### ISSUE TITLE
GCS storage failing with "invalid RSA key"

### ISSUE THREAD

**Issue by brndnmtthws** [mozilla/sccache#489]
sccache recently stopped working with GCS. I tried rolling back to a previous version, and that didn't help.

I did some digging, and found the following being printed in the logs (perhaps something changed in the Google API?):

```
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
 WARN 2019-08-04T15:12:41Z: sccache::cache::gcs: Got GCS error: Error(Msg("invalid RSA key"), State { next_error: None, backtrace: InternalBacktrace })
```

**Comment by fceller** [comment]
@brndnmtthws as a workaround until the fix is available in an official version, I got it working by converting the RSA key inside the JSON using: https://stackoverflow.com/questions/17733536/how-to-convert-a-private-key-to-an-rsa-private-key

**Comment by prettynatty** [comment]
I have the same issue when using json credentials file with `SCCACHE_GCS_KEY_PATH`. Is it fixed only for token in #578?
